### PR TITLE
smb: fix wrong endian conversion when parse NTLM Negotiate Flags - V2

### PR DESCRIPTION
**Pull request by b1tg**
Ticket: https://redmine.openinfosecfoundation.org/issues/5783

Previous PR: https://github.com/OISF/suricata/pull/8372

Changes from last PR:
    - fix invalid json schema error 
    - keep the formatting changes in a separate commit
    - fix email in commit message
  
Signed-off-by: b1tg [b1tg@protonmail.ch](mailto:b1tg@protonmail.ch)

### CODE REVIEW

**Comment by b1tg** [comment]
@victorjulien Hi, CI was failed due to network issue, how can i re-run this.

**Comment by jufajardini** [comment]
> @victorjulien Hi, CI was failed due to network issue, how can i re-run this.

I have triggered a re-run, but this might be a github actions issue for the moment. If that is the case, we may need to wait a bit ;)

**Comment by b1tg** [comment]
Seems no re-run triggered anymore🥲

**Comment by jufajardini** [comment]
> Seems no re-run triggered anymoresmiling_face_with_tear

Triggered another re-run. Let's see how it goes now, I see that the actions that were failing seem to be working with master, now...

**Comment by b1tg** [comment]
Only `CIFuzz / Fuzzing (address) (pull_request)` is in failed status now, it seems not be re-runed

**Comment by b1tg** [comment]
Any update? @jufajardini

File: rust/src/smb/auth.rs [hunk 1]
```diff
@@ -229,3 +230,30 @@ pub fn parse_secblob(blob: &[u8]) -> Option<SpnegoRequest>
...
+        let blob = hex::decode("a18202313082022da0030a0101a28202100482020c4e544c4d5353500003000000180018009c00000048014801b40000001e001e005800000008000800760000001e001e007e00000010001000fc010000158288e20a005a290000000fc6107a73184fb65fe684f6a1641464be4400450053004b0054004f0050002d0032004100450046004d003700470075007300650072004400450053004b0054004f0050002d0032004100450046004d003700470000000000000000000000000000000000000000000000000028a0c9f4e792c408913d2878feaa9a22010100000000000078a7ed218527d2010cf876f08a0b3bfa0000000002001e004400450053004b0054004f0050002d00560031004600410030005500510001001e004400450053004b0054004f0050002d00560031004600410030005500510004001e004400450053004b0054004f0050002d00560031004600410030005500510003001e004400450053004b0054004f0050002d0056003100460041003000550051000700080078a7ed218527d20106000400020000000800300030000000000000000100000000200000ad865b6d08a95d0e76a94e2ca013ab3f69c4fd945cca01b277700fd2b305ca010a001000000000000000000000000000000000000900280063006900660073002f003100390032002e003100360038002e003100390039002e003100330033000000000000000000000000005858824ec4a47b3b42ad3132ab84a5c3a31204100100000092302d756840453f00000000").unwrap();
```
Comment (jufajardini) [inline rust/src/smb/auth.rs:256]: Can you add a comment indicating what's the content of this blob buffer? :)

File: rust/src/smb/ntlmssp_records.rs [hunk 2]
```diff
@@ -154,3 +154,29 @@ pub fn parse_ntlmssp(i: &[u8]) -> IResult<&[u8], NTLMSSPRecord> {
...
     Ok((i, record))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use nom7::Err;
+    #[test]
+    fn test_parse_auth_nego_flags() {
+        let blob = [0x15, 0x82, 0x88, 0xe2];
```
Comment (jufajardini) [inline rust/src/smb/ntlmssp_records.rs:164]: Ideally, here, too, we'd like to see a comment indicating what info is here

**Review by jufajardini — CHANGES_REQUESTED** [review]
Hi, thanks for your bug report and your interest in working on this issue!

Apologies for us not giving any feedback earlier, some periods are a bit busier...

I've left two minor inline comments, and I have some more notes:
- could you please submit a PR with a Suricata-verify test to showcase this? Ideally, the test should fail now, to indicate the issue
- Can you please adjust your commit messages to follow our contribution guidelines? This includes commit message length and style; follow the author name guidelines (we need this info for licensing purposes :) )

Once these aspects are set, please submit a new PR, if it is good, then Victor or Jason will approve and then it could be merged.

I have already updated the redmine ticket and assigned it to you, to reflect the work you've already done here :)

**Comment by b1tg** [comment]
checkout https://github.com/OISF/suricata/pull/8446

**Comment by jufajardini** [comment]
Closing this in favour of new version https://github.com/OISF/suricata/pull/8449

**Pull request closed without merging**